Lib/compression/zstd/__init__.py

```diff
@@ -0,0 +1,244 @@
+    chunks = b''.join(samples)
+    chunk_sizes = tuple(_nbytes(sample) for sample in samples)
+    if not chunks:
+        raise ValueError("The samples are empty content, can't finalize dictionary.")
+    dict_content = _finalize_dict(zstd_dict.dict_content,
+                                  chunks, chunk_sizes,
+                                  dict_size, level)
+    return ZstdDict(dict_content)
+
+def compress(data, level=None, options=None, zstd_dict=None):
+    """Return Zstandard compressed *data* as bytes.
+
+    Refer to ZstdCompressor's docstring for a description of the
+    optional arguments *level*, *options*, and *zstd_dict*.
+
+    For incremental compression, use an ZstdCompressor instead.
+    """
+    comp = ZstdCompressor(level=level, options=options, zstd_dict=zstd_dict)
+    return comp.compress(data, mode=ZstdCompressor.FLUSH_FRAME)
+
+def decompress(data, zstd_dict=None, options=None):
+    """Decompress one or more frames of data.
+
+    Refer to ZstdDecompressor's docstring for a description of the
+    optional arguments *zstd_dict*, *options*.
+
+    For incremental decompression, use an ZstdDecompressor instead.
+    """
+    results = []
+    while True:
+        decomp = ZstdDecompressor(options=options, zstd_dict=zstd_dict)
+        results.append(decomp.decompress(data))
+        if not decomp.eof:
+            raise ZstdError("Compressed data ended before the "
+                            "end-of-stream marker was reached")
+        data = decomp.unused_data
+        if not data:
+            break
+    return b"".join(results)
+
+
+class CParameter(enum.IntEnum):
+    """Compression parameters"""
+
+    compressionLevel = _zstd._ZSTD_c_compressionLevel
+    windowLog = _zstd._ZSTD_c_windowLog
+    hashLog = _zstd._ZSTD_c_hashLog
+    chainLog = _zstd._ZSTD_c_chainLog
+    searchLog = _zstd._ZSTD_c_searchLog
+    minMatch = _zstd._ZSTD_c_minMatch
+    targetLength = _zstd._ZSTD_c_targetLength
+    strategy = _zstd._ZSTD_c_strategy
+
+    enableLongDistanceMatching = _zstd._ZSTD_c_enableLongDistanceMatching
+    ldmHashLog = _zstd._ZSTD_c_ldmHashLog
+    ldmMinMatch = _zstd._ZSTD_c_ldmMinMatch
+    ldmBucketSizeLog = _zstd._ZSTD_c_ldmBucketSizeLog
+    ldmHashRateLog = _zstd._ZSTD_c_ldmHashRateLog
+
+    contentSizeFlag = _zstd._ZSTD_c_contentSizeFlag
+    checksumFlag = _zstd._ZSTD_c_checksumFlag
+    dictIDFlag = _zstd._ZSTD_c_dictIDFlag
+
+    nbWorkers = _zstd._ZSTD_c_nbWorkers
+    jobSize = _zstd._ZSTD_c_jobSize
+    overlapLog = _zstd._ZSTD_c_overlapLog
+
+    def bounds(self):
+        """Return lower and upper bounds of a compression parameter, both inclusive."""
+        return _zstd._get_param_bounds(is_compress=True, parameter=self.value)
+
+
+class DParameter(enum.IntEnum):
+    """Decompression parameters"""
+
+    windowLogMax = _zstd._ZSTD_d_windowLogMax
+
+    def bounds(self):
+        """Return lower and upper bounds of a decompression parameter, both inclusive."""
```

Comment:
indicate the return type, i'm presumably a tuple (lower, upper) of ints.

Reply:
Tried to make this a bit better, hopefully the new version looks ok?